oscaralanpierce/skyrim_inventory_management_frontend
# Set up Firebase and GitHub Actions deploys

## Context

[**Set up deploy pipeline**](https://trello.com/c/D8ttijyI/235-set-up-deploy-pipeline)

We've decided to use Firebase to deploy the new front end. Firebase requires certain configurations to work, and is able to be configured to deploy with GitHub actions.

## Changes

* Install and configure Firebase
* Add dotenv to manage local environment variable configuration
* Configure deploy pipeline through GitHub Actions (deploys on merge to `main`)
* Small change to homepage (added period) to be able to visually tell if GitHub Actions has deployed to Firebase

## Considerations

I was able to successfully deploy the app from my local. In order to see whether deploys from GitHub Actions are successful, I added a period to the `App.tsx` file so I can visually tell whether the app has deployed via GitHub actions after merge to main.

## Manual Test Cases

After merging to `main` and waiting for CI to pass, visit the app's [production site](https://skyrim-inventory-management.firebaseapp.com/) in Firebase and verify that the sentence "Click on the Vite and React logos to learn more." ends with a period. If it does, that means the site has successfully deployed through GitHub actions.
